modules/querier/http.go

```diff
@@ -37,9 +42,11 @@ func (q *Querier) TraceByIDHandler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	resp, err := q.FindTraceByID(ctx, &tempopb.TraceByIDRequest{
-		TraceID: byteID,
+		TraceID:        byteID,
```

Comment:
we need to handle the case where none of these parameters are present.  default block start to `0x00...`, end to `0xFF...` and query ingesters to true.

also, how are you encoding/decoding blockstart/end key?  i don't think a direct cast of a string to bytes is what we want.  probably need to encode/decode a hex string.

we should also parse the parameters here and return an appropriate error if any parameters don't make sense.  i.e. a blockstart/end string that is not hex encoded or a query ingesters param is not "true" or "false".

Reply:
Yeah, there are two approaches to this - one is to set/reset the query parameters at the http handler level (point of entry at modules/querier/http.go) and the second is to propagate them to the lowest level - the case where none of the parameters are present is handled here - https://github.com/grafana/tempo/pull/375/files#diff-d68ed2a23a87b802e3d85870a36046be8d2f61dd2caeab632503164c9b0ab57dR251

I agree the first approach is cleaner, and we can make it generic enough to be a validation layer for the parameters.

encoding/decoding is chosen as bytes for now because it is easier to split the block range when working with raw bytes. I'm open to changing it to a guid.String() which is hex encoded. The flow at the query frontend would then be - a) generate the block shard ranges using raw bytes b) perform the following uuid.FromBytes(blockRange).String() to generate the hex encoding